Modules/socketmodule.c

```diff
@@ -3489,7 +3490,7 @@ sock_connect(PySocketSockObject *s, PyObject *addro)
     }
 
     res = internal_connect(s, SAS2SA(&addrbuf), addrlen, 1);
-    if (res < 0)
+    if (res == -1 && PyErr_Occurred())
         return NULL;
```

Comment:
With this scheme, error handling goes like this:
```suggestion
    if (res < 0) {
        assert(PyErr_Occurred());
        return NULL;
    }
```
(usually you'll see it without the assert)